homeassistant/components/zamg/weather.py

```diff
@@ -64,38 +64,47 @@ def condition(self) -> str | None:
     def native_temperature(self) -> float | None:
         """Return the platform temperature."""
         try:
-            return float(self.coordinator.data[self.station_id]["TL"]["data"])
-        except (KeyError, ValueError):
+            return float(
+                self.coordinator.data[self.station_id]["TLAM"]["data"]
+                or self.coordinator.data[self.station_id]["TL"]["data"]
```

Comment:
If the value is 0 it will be falsy, but 0 is a valid value, right?

Reply:
Yes, you are right.

Reply:
Should be fixed